lib/package/sns/validateSnsServiceProxy.test.js

```diff
@@ -0,0 +1,136 @@
+'use strict'
+
+const chai = require('chai')
+const Serverless = require('serverless/lib/Serverless')
+const AwsProvider = require('serverless/lib/plugins/aws/provider/awsProvider')
+const ServerlessApigatewayServiceProxy = require('./../../index')
+
+chai.use(require('chai-as-promised'))
+const expect = require('chai').expect
+
+describe('#validateSnsServiceProxy()', () => {
+  let serverless
+  let serverlessApigatewayServiceProxy
+
+  beforeEach(() => {
+    serverless = new Serverless()
+    serverless.servicePath = true
+    serverless.service.service = 'apigw-service-proxy'
+    const options = {
+      stage: 'dev',
+      region: 'us-east-1'
+    }
+    serverless.setProvider('aws', new AwsProvider(serverless))
+    serverless.service.provider.compiledCloudFormationTemplate = { Resources: {} }
+    serverlessApigatewayServiceProxy = new ServerlessApigatewayServiceProxy(serverless, options)
+  })
+
+  it('should validate the "topicName" property existence', async () => {
+    serverlessApigatewayServiceProxy.validated = {
+      events: [
+        {
+          serviceName: 'sns',
+          http: {
+            path: 'sns',
+            method: 'post'
+          }
+        }
+      ]
+    }
+
+    expect(() => serverlessApigatewayServiceProxy.validateSnsServiceProxy()).to.throw(
+      serverless.classes.Error
+    )
+  })
+
+  it('should validate the "topicName" property if it is string or AWS intrinsic function', async () => {
+    serverlessApigatewayServiceProxy.validated = {
+      events: [
+        {
+          serviceName: 'sns',
+          http: {
+            topicName: ['xx', 'yy'],
+            path: 'sns',
+            method: 'post'
+          }
+        }
+      ]
+    }
+
+    expect(() => serverlessApigatewayServiceProxy.validateSnsServiceProxy()).to.throw(
+      serverless.classes.Error
+    )
+  })
+
+  it('should validate the "topicName" property that AWS intrinsic function is correct syntax', async () => {
```

Comment:
similarly, here, the test is really 'it should throw if topicName is an intrinsic function that's not Fn::GetAtt'

Reply:
I agree again :)